### PR TITLE
fripp: misc cleanup

### PR DESCRIPTION
**Pull request by garlick**
Fixing some minor nits with the fripp stats collection code, encountered while  instrumenting #3822.

(Apologies to @garrettbslone for not catching some of these during the review window!)

### CODE REVIEW

File: src/common/libflux/fripp.c [hunk 1]
```diff
@@ -341,84 +368,107 @@ static void timer_cb (flux_reactor_t *r,
...
+        int saved_errno = errno;
         flux_watcher_destroy (ctx->w);
-    }
-    if (ctx->metrics) {
-        zhashx_purge (ctx->metrics);
-        zhashx_destroy (&ctx->metrics);
-    }
-    if (ctx->done) {
+        if (ctx->metrics) {
+            zhashx_purge (ctx->metrics);
```
Comment (grondo) [inline src/common/libflux/fripp.c:393]: It appears `zhashx_destroy()` internally calls purge (minus the attempt to shrink the hash table), so perhaps this line could be removed? (Sorry I missed that in the review of the initial fripp pr)

File: src/common/libflux/fripp.c [hunk 2]
```diff
@@ -341,84 +368,107 @@ static void timer_cb (flux_reactor_t *r,
...
 
-    if ((ctx->sock = socket(AF_INET, SOCK_DGRAM, 0)) == -1) {
-        flux_log_error (h, "error opening socket");
+    if (!(ctx = calloc (1, sizeof (*ctx))))
         goto error;
+    ctx->sock = -1;
+    /* If envirnoment variable is unset, or it is set wrong, let the context
+     * be created with disabled status so we don't repeatedly try to create
+     * it "on demand" in the flux_stats_*() functions.
+     */
```
Comment (grondo) [inline src/common/libflux/fripp.c:459]: Nice! :+1:

**Review by grondo — APPROVED** [review]
LGTM! One suggestion inline, but that one is optional.

File: src/common/libflux/fripp.c [hunk 3]
```diff
@@ -341,84 +368,107 @@ static void timer_cb (flux_reactor_t *r,
...
+    int e;
+
+    if (!(node = strdup (s))) {
+        snprintf (errbuf, errbufsz, "out of memory");
+        return -1;
     }
-    if (!(host = calloc (1, port_s - uri + 1))) {
-        flux_log_error (h, "fripp_ctx_create");
+    if (!(service = strchr (node, ':'))) {
+        snprintf (errbuf, errbufsz, "missing colon delimiter");
```
Comment (grondo) [inline src/common/libflux/fripp.c:422]: Looks like tests of malformed `FLUX_FRIPP_STATSD` could be added (missing `:` and failure from `getaddrinfo()` below) to give better coverage of these conditions, but that is optional IMO.

**Comment by garlick** [comment]
OK - addressed your suggestions @grondo, ans since there's already approval, I'll set MWP.

**Pull request merged**